test/Tereshchenko_Igor/laboratory1.sql

```diff
@@ -1,2 +1,151 @@
+  
+/* --------------------------------------------------------------------------- 
+3. Надати додаткові права користувачеві (створеному у пункті № 1) для створення таблиць, 
+внесення даних у таблиці та виконання вибірок використовуючи команду ALTER/GRANT. 
+Згенерувати базу даних використовуючи код з теки OracleScript та виконати запити: 
+---------------------------------------------------------------------------*/
+--Код відповідь:
+GRANT INSERT ANY TABLE TO tereshchenko ;
+
+
+/*---------------------------------------------------------------------------
+3.a. Вивести дані про покупця, що проживає в Америці: його ім’я, назвавши стовпчик як client_name,
+а його ZIP-код - назвавши цей стовпчик client_code напроти кожного покупця вивести номер його замовлення. 
+Join використовувати заборонено. Виконати завдання в SQL. 4 бали
+---------------------------------------------------------------------------*/
+
+--Код відповідь:
+
+SELECT                        -- select customers with orders
+  CUST_NAME as "client_name", -- save register using " "
+  CUST_ZIP as "client_code",
+  ORDER_NUM
+FROM CUSTOMERS, ORDERS
+  WHERE
+    CUST_COUNTRY = 'USA'
+    and
+    CUSTOMERS.CUST_ID=ORDERS.CUST_ID
+    
+UNION                         -- union customers without order
+
+SELECT
+  CUST_NAME as "client_name", -- save register using " "
+  CUST_ZIP as "client_code",
+  null as ORDER_NUM           -- rename column to make UNION
+FROM CUSTOMERS
+  WHERE CUST_ID NOT IN (
+                        SELECT CUST_ID   --CUST_ID of customers with orders
+                          FROM ORDERS
+                      )
+    and
+    CUST_COUNTRY = 'USA';
+
+
+/*---------------------------------------------------------------------------
+3.b. Вивести дані покупця: його ім’я, назвавши client_name заголовок стовпчика та 
+повну його адресу як одне поле, взявши zip код у дужки –  назва стовпчика client_address. 
+Виконати завдання в SQL. 4 бали
+---------------------------------------------------------------------------*/
+
+--Код відповідь:
+SELECT 
+  TRIM(CUST_NAME) as "client_name", -- save register using " "
+  TRIM(CUST_ADDRESS) 
+  ||' '
+  || TRIM(CUST_CITY) 
+  || TRIM(CUST_STATE) 
+  || ' (' 
+  || TRIM(CUST_ZIP) 
+  || ') ' 
+  || TRIM(CUST_COUNTRY) as "client_address"
+FROM CUSTOMERS;
+
+/*---------------------------------------------------------------------------
+c. Вивести назву товару* на ім’я замовника, який його не замовляв. 
+Виконати завдання в SQL. 4 бали
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+SELECT                      -- final projection, we don't need CUST_ID
+  TRIM(PROD_NAME),
+  TRIM(CUST_NAME)
+FROM (
+        SELECT              -- select all combination of customer and product
+          CUSTOMERS.CUST_ID,
+          CUST_NAME,
+          PROD_NAME
+        FROM CUSTOMERS, PRODUCTS 
```

Comment:
Full 

- [ ] fields name

- [ ] distinct

Reply:
Fixed

Reply:
Not all fixed

Reply:
Sorry fixed

Reply:
Fixed

Reply:
done

Reply:
dgddgdgd